stock_move_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+    actual_date_source = fields.Date(
+        help="Technical field to store the actual_date of the source document."
+    )
+
+    def _get_timezone(self):
+        return self.env.context.get("tz") or self.env.user.tz or "UTC"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # This handles the case where a move is created separately after the parent record.
+        # For example, in mrp_stock_actual_date, the actual_date is passed via context
+        # when validating an unbuild order or a scrap.
+        res = super().create(vals_list)
+        actual_date_source = self.env.context.get("actual_date_source")
+        if actual_date_source:
+            for rec in res:
```

Comment:
In v16, you can do :

```suggestion
           res.actual_date_source = actual_date_source
```

And even if Odoo does write `res` often, I would have used meaningful name instead like `moves` or `created_moves`